src/managers/GuildEmojiRoleManager.js

```diff
@@ -53,13 +53,17 @@ class GuildEmojiRoleManager {
   add(roleOrRoles) {
     if (roleOrRoles instanceof Collection) return this.add(roleOrRoles.keyArray());
     if (!Array.isArray(roleOrRoles)) return this.add([roleOrRoles]);
-    roleOrRoles = roleOrRoles.map(r => this.guild.roles.resolve(r));
 
-    if (roleOrRoles.includes(null)) {
-      return Promise.reject(new TypeError('INVALID_TYPE', 'roles', 'Array or Collection of Roles or Snowflakes', true));
+    const resolvedRoles = [];
+    for (const role of roleOrRoles) {
```

Comment:
```suggestion
    for (const role of roleOrRoles.values()) {
```
Otherwise passing a collection will error

Reply:
A collection will never reach this code as is right now, should I remove  the recursive call and use .values() here?

Reply:
Oh, didnt notice that, removing the recursive call would probably be better